packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -3,50 +3,48 @@ package expo.modules.image
 import android.util.Log
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.model.GlideUrl
-import com.facebook.react.bridge.Promise
-import com.facebook.react.bridge.ReactApplicationContext
-import com.facebook.react.bridge.ReactContextBaseJavaModule
-import com.facebook.react.bridge.ReactMethod
+import expo.modules.core.errors.ModuleDestroyedException
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.launch
-import java.lang.Exception
-import java.lang.IllegalStateException
-import java.util.concurrent.CancellationException
 
-class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+class ExpoImageModule : Module() {
   private val moduleCoroutineScope = CoroutineScope(Dispatchers.IO)
-  override fun getName() = "ExpoImageModule"
 
-  @ReactMethod
-  fun prefetch(url: String, promise: Promise) {
-    moduleCoroutineScope.launch {
-      try {
-        val glideUrl = GlideUrl(url)
-        val result = Glide.with(context)
+  override fun definition() = ModuleDefinition {
+    Name("ExpoImageModule")
```

Comment:
What would you say if we have just one `ExpoImage` module with functions + the view? That's what we have on iOS right now.